Updated Intro to Zoe to Alpha. (#274)
Agoric/documentation: main/getting-started/intro-zoe.md

just descriptions of assets, and not the actual assets. The actual assets are contained in ERTP `purses` and `payments`, not `amounts`.

She then creates her proposal, using a record with `give`, `want`, and `exit` (optional) properties. `give` and `want` are `KeywordAmount` records with
keywords as keys and `amounts` as values. `exit` is an `ExitRule` record specifying how/when a user can exit the contract. In the above,
Alice wants at least 25 Moola, in exchange for giving at most 100 Quatloos, where she can exit the offer at any time just by
asking, and get her assets back (or get her `want` value if the offer happened to be satisfied before then).

The `harden()` command is how we build a defensible API surface around an object by freezing all reachable properties. It’s similar,
but not identical, to JavaScript’s `Object.freeze`. For more information on `harden`, see [here](https://www.npmjs.com/package/@agoric/harden#background-why-do-we-need-to-harden-objects)

Now, Alice is ready to use her `invitation`, `proposal`, and `payment` to make an offer and participate in the Atomic Swap
contract `instance`. 

```js

> const { outcome, payout: alicePayoutP } = await E(zoe).offer(

Reviewer: Replace example with 
const userSeat = E(zoe).offer(
  aliceInvite,
  aliceProposal,
  alicePayments,
);
const outcome = await E(userSeat).getOfferResult();
const payouts = await E(userSeat).getPayouts();

  alicePayments,
);
```

> Zoe checks the invitation for its validity for that contract instance. If it’s an invalid invitation, the offer attempt fails, and Alice gets

Reviewer: Zoe checks the invitation for its validity for that contract instance. When 
she makes her offer, Alice receives a (promise for) a `UserSeat` that lets
her monitor or control her offer, including getting payouts or offer results, 
or exiting it altogether. If her offer was invalid, the 
offer attempt fails, and Alice gets her refund in the payout. If the
offer is valid, it's now an *active offer*.

Now, Alice needs to get someone else involved who potentially will also 
make an offer, hopefully one that offers what she wants for
the price she’s willing to pay for it. The "offer result" is the object 
returned by the contract when it processes an offer. In the case of 
`atomicSwap`, it used `zcf.makeInvitation()` to make and return an *invitation* Alice can 
send to others, in this case Bob.

> her refund in the payout. When she makes her offer, Alice receives a *promise* that resolves to her *payout*. If the

Reviewer: NO. She gets a UserSeat from which she can get the payout

  alicePayments,
);
```
Zoe checks the invitation for its validity for that contract instance. If it’s an invalid invitation, the offer attempt fails, and Alice gets
her refund in the payout. When she makes her offer, Alice receives a *promise* that resolves to her *payout*. If the
offer is valid, it's now an *active offer*.

Now, Alice needs to get someone else involved who potentially will also make an offer, hopefully one that offers what she wants for
the price she’s willing to pay for it. Alice uses `zcf.makeInvitation()` to make an *invitation* she can 

> send to others, in this case Bob.

Author: Interesting. I hadn't grokked the need for the contract to give users the power to get more invitations, and I don't think that's expressly stated anywhere and it should be. (feel free to correct me if it is stated somewhere).

> send to others, in this case Bob.

Reviewer: replaced above

```

Bob decides to make his own offer, which happens to match up with Alice’s offer (assume his payments were constructed similar to Alice’s);
```const bobProposal = harden({
  want: { Asset: quatloos100 },
  give: { Price: moola25 },
  exit: { onDemand: null },
})

// Bob escrows with zoe and makes an offer
const { outcome: bobOfferResult, payout: bobPayoutP } = await E(zoe).offer(
  bobExclusiveInvitation,
  bobProposal,
  bobPayments,
);
```
Bob has also gotten back a *promise* for a *payout*. His offer is also now an *active offer*.

## Satisfying and completing offers

At this point, both the offers that Alice and Bob made are active and known to the contract instance. The Atomic Swap contract 
code determines that they are matching offers.  The contract instance calls `reallocate()`, which *reallocates* the amounts which 
are the accounting records within Zoe. Payouts are not created here. 

The contract instance then *exits* the offers. **tyg todo: Not quite sure what the current implementation is here** makes the payouts to
the offer holders.  This takes the amounts from the account records, and withdraws the offer-specified amounts from the 

> digital assets escrowed within Zoe. This is when 

Author: Yeah, when this was originally written, we didn't have swap() or trade(). Now, I sorta want either leave them out with a note "You should take a look at the swap() and trade() helpers here" or very briefly cover them.

Still the question of whether "exits the offers and makes the payouts" is the right phrasing/nomenclation/procedure these days?

</td>
</tr>
<tr>

> <td colspan="2">At this point, everything is installed and set up and you’ve got a contract instance. The following rows cover what happens during
> an Atomic Swap contract instance.

Reviewer 2: ```suggestion
<td colspan="2">At this point, everything is installed and set up and you’ve got a contract instance. The following rows cover what happens while
the Atomic Swap contract instance is active.
```

</td>
<tr>
<tr>
<td><center>1</center></td>
<td>A holder of an invitation to this contract instance uses it to get a seat. The assets used in the offer are escrowed with Zoe.
</td>
</tr>
<tr>
<td><center>2</center></td>
<td>Likely via means outside of the contract, invitations are distributed to potential users of this contract instance).</td>
</tr>
<tr>
<td><center>3</center></td>
<td>Recipients use the InviteIssuer to claim and validate their invitation.
</td>
</tr>
<tr>
<td><center>4</center></td>
<td>Invitation holders make their offers.
</td>
</tr>
<tr>
<td><center>5</center></td>

> <td>The contract instance code processes the offers, which may request some reallocations as a result.</td>

Reviewer 2: ```suggestion
<td>The contract code processes the offers, which may request some reallocations as a result.</td>
```
